airbyte-webapp/src/packages/cloud/components/experiments/FreeConnectorProgram/FreeConnectorProgramEnrollmentButton/EnrollmentModal.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from "react";
+import { FormattedMessage } from "react-intl";
+
+import { Button, ButtonVariant } from "components/ui/Button";
+import { FlexContainer, FlexItem } from "components/ui/Flex";
+import { Heading } from "components/ui/Heading";
+import { ModalFooter } from "components/ui/Modal/ModalFooter";
+
+import { useModalService } from "hooks/services/Modal";
+import { StripeCheckoutSessionCreate, StripeCheckoutSessionRead } from "packages/cloud/lib/domain/stripe";
+import { useStripeCheckout } from "packages/cloud/services/stripe/StripeService";
+import { useCurrentWorkspaceId } from "services/workspaces/WorkspacesService";
+
+import { CardSVG } from "./CardsSvg";
+import { ConnectorGridSvg } from "./ConnectorGridSvg";
+import styles from "./EnrollmentModal.module.scss";
+import { FreeSVG } from "./FreeSvg";
+import { MailSVG } from "./MailSvg";
+
+const STRIPE_SUCCESS_QUERY = "stripeCheckoutSuccess";
+
+interface EnrollmentModalContentProps {
+  closeModal: () => void;
+  createCheckout: (p: StripeCheckoutSessionCreate) => Promise<StripeCheckoutSessionRead>;
+
+  workspaceId: string;
+}
+
+const EnrollmentModalContent: React.FC<EnrollmentModalContentProps> = ({ closeModal, createCheckout, workspaceId }) => {
+  const [isLoading, setIsLoading] = useState(false);
+
+  const startStripeCheckout = async () => {
+    setIsLoading(true);
+    // Use the current URL as a success URL but attach the STRIPE_SUCCESS_QUERY to it
+    const successUrl = new URL(window.location.href);
+    successUrl.searchParams.set(STRIPE_SUCCESS_QUERY, "true");
+    const { stripeUrl } = await createCheckout({
+      workspaceId,
+      successUrl: successUrl.href,
+      cancelUrl: window.location.href,
+      stripeMode: "setup",
+    });
+
+    // Forward to stripe as soon as we created a checkout session successfully
+    window.location.assign(stripeUrl);
+  };
```

Comment:
Largely copied from our credit purchasing flow [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/RemainingCredits.tsx#L104).

After successfully entering a card, users will be redirected back to the URL they came from with an extra query param: `?stripeCheckoutSuccess=true`